### PR TITLE
Display latest sync time for pull mirrors on the repo page

### LINKED ISSUE #25168: Mirror Synchronization status display
### Feature Description

Synchronization status display

### Screenshots

截图_选择区域_20230609112620
截图_选择区域_20230609162654

### PR DESCRIPTION
**Pull request by dsseng**
Fixes #25168

### CODE REVIEW

File: models/repo/repo.go [hunk 1]
```diff
@@ -189,6 +189,7 @@ type Repository struct {
 
 	CreatedUnix  timeutil.TimeStamp `xorm:"INDEX created"`
 	UpdatedUnix  timeutil.TimeStamp `xorm:"INDEX updated"`
+	SyncedUnix   timeutil.TimeStamp `xorm:"DEFAULT 0"`
```
Comment (dsseng) [inline models/repo/repo.go]: Maybe it'd be better if we add PullSyncedAt and PushSyncedAt as we change this

**Comment by techknowlogick** [comment]
Could you provide a screenshot?

File: templates/repo/header.tmpl [hunk 2]
```diff
@@ -27,7 +27,11 @@
 					</div>
 				</div>
 				{{if $.PullMirror}}
-					<div class="fork-flag">{{ctx.Locale.Tr "repo.mirror_from"}} <a target="_blank" rel="noopener noreferrer" href="{{$.PullMirror.RemoteAddress}}">{{$.PullMirror.RemoteAddress}}</a></div>
+					<div class="fork-flag">
+						{{ctx.Locale.Tr "repo.mirror_from"}}
+						<a target="_blank" rel="noopener noreferrer" href="{{$.PullMirror.RemoteAddress}}">{{$.PullMirror.RemoteAddress}}</a>
+						{{ctx.Locale.Tr "repo.mirror_sync"}} {{TimeSinceUnix .SyncedUnix ctx.Locale}}
```
Comment (delvh) [inline templates/repo/header.tmpl]: I don't understand your implementation at all.
All this data is already stored.
You simply need to call `{{TimeSinceUnix $.PullMirror.UpdatedUnix ctx.Locale}}`

**Review by delvh — DISMISSED** [review]

File: templates/repo/header.tmpl (hunk 2 above)
Comment (dsseng) [reply templates/repo/header.tmpl]: Updated is being set to latest commit date received, not the date of sync itself. At least this is what I got from the code.

Users on the other hand typically want to know mirror they refer to has been recently synced to upstream

Comment (delvh) [reply templates/repo/header.tmpl]: No, I just had a look at the code, `UpdatedUnix` is set to the time of the sync.
https://github.com/go-gitea/gitea/blob/c5901b622dd5fae8da336b94150e63292489afed/services/mirror/mirror_pull.go#L409

Comment (dsseng) [reply templates/repo/header.tmpl]: That's about setting updated time on a mirror object. Line 546 sets repository update time to the latest commit date (and from the template we access the repository object)

**Comment by sh7dm** [comment]
Screenshots:

image
image

File: templates/repo/header.tmpl (hunk 2 above)
Comment (delvh) [reply templates/repo/header.tmpl]: Yes, but the pull mirror is also available in the template (see my initial message at the location where I originally commented).
I'm strongly against adding this field to `repository`:
That field should belong to a pull mirror (object), not just any repository.
Even more so as you modify the API with your approach, and I don't want to pollute the API with information at the wrong place.

Comment (dsseng) [reply templates/repo/header.tmpl]: Alright, thanks, will take a look

**Comment by lunny** [comment]
`Updated` time maybe zero.

File: templates/repo/header.tmpl [hunk 3]
```diff
@@ -27,7 +27,11 @@
 					</div>
 				</div>
 				{{if $.PullMirror}}
-					<div class="fork-flag">{{ctx.Locale.Tr "repo.mirror_from"}} <a target="_blank" rel="noopener noreferrer" href="{{$.PullMirror.RemoteAddress}}">{{$.PullMirror.RemoteAddress}}</a></div>
+					<div class="fork-flag">
+						{{ctx.Locale.Tr "repo.mirror_from"}}
+						<a target="_blank" rel="noopener noreferrer" href="{{$.PullMirror.RemoteAddress}}">{{$.PullMirror.RemoteAddress}}</a>
+						{{ctx.Locale.Tr "repo.mirror_sync"}} {{TimeSinceUnix $.PullMirror.UpdatedUnix ctx.Locale}}
```
Comment (delvh) [inline templates/repo/header.tmpl]:
```suggestion
						{{if $.PullMirror.UpdatedUnix}}{{ctx.Locale.Tr "repo.mirror_sync"}} {{TimeSinceUnix $.PullMirror.UpdatedUnix ctx.Locale}}{{end}}
```

**Review by delvh — APPROVED** [review]
LGTM after that small improvement

File: templates/repo/header.tmpl (hunk 3 above)
Comment (dsseng) [reply templates/repo/header.tmpl]: Thank you!

**Review by lunny — APPROVED** [review]

**Pull request merged**